File: Java/2-Programs-based-on-Numbers/28-hexadecimal-to-decimal-conversion.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner getInput = new Scanner(System.in);

        System.out.println("Enter Hexadecimal number:");
        String hexa = getInput.next().toUpperCase();

        int decimal = 0;
        int power = 0;

        for (int i = hexa.length()-1; i>=0; i--) {
            char ch = hexa.charAt(i);
            int value;

            if (ch >= '0' && ch <= '9') {
                value = ch - '0';
            } else {
                value = ch - 'A'+10;
            }
            decimal += value * (int)Math.pow(16, power);
            power++;
        }
        System.out.println("Decimal value is: " + decimal);
        getInput.close();
    }
}

File: Python/2-Programs-based-on-Numbers/28-hexadecimal-to-decimal-conversion.py
hexa = input("Enter Hexadecimal number: ").upper()

decimal = 0
power = 0

for digit in hexa[::-1]:
    if '0' <= digit <= '9':
        value = int(digit)
    else:
        value = ord(digit) - ord('A')+10

    decimal += value * (16 ** power)
    power += 1

print("Decimal value is:", decimal)


# Using built-in method
# decimal = int(hexa, 16)